### PR TITLE
feat(content): weather reports from radio, new flag WEATHER_FORECAST

### PR DESCRIPTION
**Pull request by Lamandus**
## Purpose of change
There was a weather report, which was... a bit on the bad side. It worked, but didn't look good. To get it, you needed a radio and the right broadcasting tower, which was a chore. Now you can activate your radio and directly read out the weather report of the nearest broadcasting tower.

It uses a new flag "FORECAST_WEATHER" and can be used with the use_action "WEATHER_TOOL". Right now, only the radio has the possibility. It can be further expanded, if needed (for example, most MP3 players have a inbuilt FM/AM Radio)

## Testing
Took some time and @chaosvolt helped a lot!

## Additional context
grafik
grafik

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
Sometimes radio fails to get weather information. In this way it does not produce any sensible messages. Could you add something like: "No signal from weather station"?

Also could you add info to radio message about what day of the week it is now?

**Comment by Lamandus** [comment]
> Sometimes radio fails to get weather information. In this way it does not produce any sensible messages. Could you add something like: "No signal from broadcasting tower" or "Broadcasting weather signal was not found"?
> 
> It can be added later if check like that is hard to implement.

I am totally honest with you, that would totally impossible for me to do. This alone was hard (I never learned C++, all that stuff is just reverse engeneering)

**Comment by Firestorm01X2** [comment]
> I am totally honest with you, that would totally impossible for me to do. This alone was hard (I never learned C++, all that stuff is just reverse engeneering)

No worries. Can be done later in other PRs.

**Review by Firestorm01X2 — DISMISSED** [review]
Ready.

File: src/iuse.cpp [hunk 1]
```diff
@@ -9023,7 +9023,17 @@ int iuse::weather_tool( player *p, item *it, bool, const tripoint & )
                                   print_pressure( static_cast<int>( weatherPoint.pressure ) ) );
         }
     }
-
+    if( it->has_flag( flag_WEATHER_FORECAST ) ) {
+        std::string message = string_format( "", message );
+        const auto tref = overmap_buffer.find_radio_station( it->frequency );
+        if( tref ) {
+            const auto selected_tower = tref.tower;
```
Comment (Firestorm01X2) [inline src/iuse.cpp:9030]: This variable is not used. It must be removed

**Review by chaosvolt — APPROVED** [review]
Hopin' we gud

**Pull request merged**